pkg/minishift/network/settings.go

```diff
@@ -43,31 +57,254 @@ type NetworkSettings struct {
+	networkSettings := GetNetworkSettingsFromInstance(driver)
+
+	// VirtualBox and KVM rely on two interfaces
+	// eth0 is used for host communication
+	// eth1 is used for the external communication
+	if minishiftConfig.IsVirtualBox() {
+		dhcpNetworkSettings := NetworkSettings{
+			Device:  "eth0",
+			UseDHCP: true,
+		}
+		WriteNetworkSettingsToInstance(driver, networkSettings)
+		WriteNetworkSettingsToInstance(driver, dhcpNetworkSettings)
+	}
+
+	// HyperV and Xhyve rely on a single interface
+	// eth0 is used for hpst and external communication
+	// eth1 is disabled
+	if minishiftConfig.IsHyperV() || minishiftConfig.IsXhyve() {
+		disabledNetworkSettings := NetworkSettings{
+			Device:   "eth1",
+			Disabled: true,
+		}
+		WriteNetworkSettingsToInstance(driver, networkSettings)
+		WriteNetworkSettingsToInstance(driver, disabledNetworkSettings)
+	}
+
+	printNetworkSettings(networkSettings)
+
+	fmt.Println(configureRestartNeededMessage)
 }
 
+// printNetworkSettings will print to stdout the values from the struct
+// NetworkSettings.
 func printNetworkSettings(networkSettings NetworkSettings) {
 	fmt.Println(configureIPAddressMessage)
 	fmt.Println("   Device:     ", networkSettings.Device)
 	fmt.Println("   IP Address: ", fmt.Sprintf("%s/%s", networkSettings.IPAddress, networkSettings.Netmask))
 	if networkSettings.Gateway != "" {
 		fmt.Println("   Gateway:    ", networkSettings.Gateway)
 	}
-	if networkSettings.DNS1 != "" {
+	if networkSettings.DNS1 != "" || networkSettings.DNS2 != "" {
 		fmt.Println("   Nameservers:", fmt.Sprintf("%s %s", networkSettings.DNS1, networkSettings.DNS2))
 	}
 }
 
+// fillNetworkSettings will populate the network configuration file for use
+// by the minishift-set-ipaddress script inside the VM. It takes the struct
+// NetworkSettings containing the values and retuns a string containing
+// a multiline config file.
 func fillNetworkSettingsScript(networkSettings NetworkSettings) string {
 	result := &bytes.Buffer{}
 
-	tmpl, err := template.New("networkScript").Parse(configureNetworkScriptTemplate)
-	if err != nil {
-		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating network script template: %s", err.Error()))
+	tmpl := template.New("networkScript")
+
+	if networkSettings.Disabled {
+		tmpl.Parse(configureNetworkScriptDisabledAddressTemplate)
+	} else if networkSettings.UseDHCP {
+		tmpl.Parse(configureNetworkScriptDynamicAddressTemplate)
+	} else {
+		tmpl.Parse(configureNetworkScriptStaticAddressTemplate)
 	}
-	err = tmpl.Execute(result, networkSettings)
+
+	err := tmpl.Execute(result, networkSettings)
 	if err != nil {
-		atexit.ExitWithMessage(1, fmt.Sprintf("Error executing network script template:: %s", err.Error()))
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error executing network script template: %s", err.Error()))
 	}
 
 	return result.String()
 }
+
+func executeCommandOrExit(driver drivers.Driver, command string, errorMessage string) string {
```

Comment:
IMO, this can be a general utility function as part of `minishift/pkg/util`.

Reply:
as a refactor... I already have to move several functions from the port check for sshfs to the new network namespace. ain't gonna happen now

Reply:
@gbraad Can we track this in a separate issue? Are you planning to create follow up issues after we merge this PR?